include/pybind11/pytypes.h

```diff
@@ -1282,6 +1282,21 @@ inline size_t len(handle h) {
     return (size_t) result;
 }
 
+inline size_t len_hint(handle h) {
+#if PY_VERSION_HEX >= 0x03040000
+    ssize_t result = PyObject_LengthHint(h.ptr(), 0);
+#else
+    ssize_t result = PyObject_Length(h.ptr());
```

Comment:
Falling back to a more accurate `PyObject_Length()` is questionable. How about this:

- >=Python3.4, everything is fine just do what you're doing now.
- <Python3.4, there is no `PyObject_Length()`, so just `return 0`.

Reply:
I assume you mean <Python3.4 there is no PyObject_LengthHint?

Reply:
Right, my bad.